test/core/event_engine/fuzzing_event_engine/fuzzing_event_engine.cc

```diff
@@ -14,25 +14,97 @@
 
 #include "test/core/event_engine/fuzzing_event_engine/fuzzing_event_engine.h"
 
+#include <inttypes.h>
+
 #include <chrono>
 
+#include <grpc/grpc.h>
+
+#include "src/core/lib/gprpp/time.h"
+#include "test/core/event_engine/fuzzing_event_engine/fuzzing_event_engine.pb.h"
+
+extern gpr_timespec (*gpr_now_impl)(gpr_clock_type clock_type);
+
 namespace grpc_event_engine {
 namespace experimental {
 
 namespace {
 const intptr_t kTaskHandleSalt = 12345;
-}
+FuzzingEventEngine* g_fuzzing_event_engine = nullptr;
+}  // namespace
 
 FuzzingEventEngine::FuzzingEventEngine(Options options)
     : final_tick_length_(options.final_tick_length) {
-  for (const auto& delay : options.actions.tick_lengths()) {
-    tick_increments_[delay.id()] += std::chrono::microseconds(delay.delay_us());
+  GPR_ASSERT(g_fuzzing_event_engine == nullptr);
+  g_fuzzing_event_engine = this;
+
+  gpr_now_impl = GlobalNowImpl;
+
+  Restart(fuzzing_event_engine::Actions());
+}
+
+void FuzzingEventEngine::Restart(const fuzzing_event_engine::Actions& actions) {
+  grpc_core::MutexLock lock(&mu_);
+
+  tick_increments_.clear();
+  task_delays_.clear();
+  tasks_by_id_.clear();
+  tasks_by_time_.clear();
+  next_task_id_ = 1;
+  current_tick_ = 0;
+  now_ = Time() + std::chrono::seconds(5);
```

Comment:
Why is `now_` 5 seconds in the future? Probably worth a comment here explaining it.

Reply:
commented